### PR TITLE
5.) refactoring localMaxPeakFinder/osmFIsh

### PR DESCRIPTION
**Pull request by shanaxel42**
This PR:
- adds a new version of LocalMaxPeakFinder to the FindSpots module that processes SpotFindingResults
-  Refactors osmFish to use FindSpots.LocalMaxPeakFinder and DecodeSpots.PerRoundMaxChannel with a trace building strategy of SEQUENTIAL

depends on: #1593  #1592 #1518 #1517

### CODE REVIEW

File: starfish/core/spots/FindSpots/local_max_peak_finder.py [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+    measurement_type : str, {'max', 'mean'}
+        default 'max' calculates the maximum intensity inside the object
+    min_num_spots_detected : int
+        When fewer than this number of spots are detected, spot searching for higher threshold
+        values. (default = 3)
+    is_volume : bool
+        Not supported. For 3d peak detection please use TrackpyLocalMaxPeakFinder.
+        (default=False)
+    verbose : bool
+        If True, report the percentage completed (default = False) during processing
```
Comment (ttung) [inline starfish/core/spots/FindSpots/local_max_peak_finder.py]: inconsistent depiction of defaults.

File: starfish/core/spots/FindSpots/local_max_peak_finder.py [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+                indices=True,
+                num_peaks=np.inf,
+                footprint=None,
+                labels=None
+            )
+
+            # stop spot finding when the number of detected spots falls below min_num_spots_detected
+            if len(spots) <= self.min_num_spots_detected:
+                stop_threshold = threshold
+                if self.verbose:
```
Comment (ttung) [inline starfish/core/spots/FindSpots/local_max_peak_finder.py]: possibly out of the scope of this PR, but this test of verbosity differs from the above one.

File: starfish/core/spots/FindSpots/local_max_peak_finder.py [hunk 3]
```diff
@@ -0,0 +1,295 @@
...
+
+        Parameters
+        ----------
+        img: Union[np.ndarray, xr.DataArray]
+            data array in which spots should be detected and over which to compute different
+            intensity thresholds
+
+        Returns
+        -------
+        Number :  #TODO ambrosejcarr this should probably be a float
```
Comment (ttung) [inline starfish/core/spots/FindSpots/local_max_peak_finder.py]: so, which is it, @ambrosejcarr?

Comment (ambrosejcarr) [reply starfish/core/spots/FindSpots/local_max_peak_finder.py]: float.

**Pull request merged**